### PR TITLE
CP-1336 kasperpeulen: make sure that transformer only touches file that need to be rewritten

### PR DESCRIPTION
**Pull request by evanweible-wf**
Copy of #27, but rebased against the latest master.

@kasperpeulen looks like I didn't commit the merge quite right so #27 still shows conflicts. I've rebased your branch against the latest master here in this PR.

This should work now @trentgrover-wf @maxwellpeterson-wf @jayudey-wf

### CODE REVIEW

**Comment by kasperpeulen** [comment]
LGTM

Thanks for the effort!

File: test/integration_test.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    final testHtml = new File('build/example/index.html');
+    final content = await testHtml.readAsString();
+
+    File expectedHtml = new File('test/expected.html');
+    expect(content, await expectedHtml.readAsString());
+  });
+}
+
+// TODO maybe we should expect the html to be formatted?
+final expectContent = '''
```
Comment (trentgrover-wf) [inline test/integration_test.dart]: doesn't look like this is still used

**Comment by trentgrover-wf** [comment]
got some dart_dev failures...

File: test/integration_test.dart (hunk 1 above)
Comment (kasperpeulen) [inline test/integration_test.dart]: you are right, I use the file for this, I will remove

**Comment by evanweible-wf** [comment]
@trentgrover-wf fixed

**Comment by trentgrover-wf** [comment]
+1

**Comment by maxwellpeterson-wf** [comment]
+1

**Comment by maxwellpeterson-wf** [comment]
@jayudey-wf ready for merge

**Comment by jayudey-wf** [comment]
QA Resource Approval: +10

- [ ] Testing instruction
- [ ] Dev +1's
- [ ] Dev/QA +10 with detail of what was tested
 - passing CI, verified that example is still working as expected
- [ ] Unit test created/updated
- [ ] All unit tests pass

Merging into master.

**Pull request merged**